pkg/vrpqstatus/status.go

```diff
@@ -65,6 +65,14 @@ func Update(ctx context.Context, clnt client.Client, log logr.Logger, vrpq *vapi
 // be updated with the status condition.
 func UpdateConditionAndState(ctx context.Context, clnt client.Client, log logr.Logger,
 	vrpq *vapi.VerticaRestorePointsQuery, condition *metav1.Condition, state string) error {
+	// Check if the condition is already set
```

Comment:
I am confused by this change. If I recall, this was done because we would endlessly loop by updating the status condition continuously. One thing I don't understand is that once the query is done, we should skip the reconcile. Wouldn't that avoid updating the status condition?

Also, this is a large PR. I would suggest pulling this out and handling it separately.